lib/ansible/modules/network/ios/ios_logging.py

```diff
@@ -297,7 +298,7 @@ def map_params_to_obj(module, required_if=None):
 
         if module.params['dest'] == 'buffered':
             if not module.params['size']:
-                module.params['size'] = str(4096)
+                module.params['size'] = int(4096)
```

Comment:
We cannot convert the value to `int` here as this would cause idempotence issue.
Rather you can convert the value to `int` in the function `validate_size` itself to make the default value work.
This would be the consistent way to make it work.
```
 def validate_size(value, module):
     if value:
-        if not int(4096) <= value <= int(4294967295):
+        if not int(4096) <= int(value) <= int(4294967295):
```